stock_owner_restriction/models/stock_quant.py

```diff
@@ -25,20 +26,24 @@ def _gather(
             owner_id=owner_id,
             strict=strict,
         )
-        restricted_owner_id = self.env.context.get("force_restricted_owner_id", None)
-        if owner_id is None or restricted_owner_id is None:
+        restricted_owner = self.env.context.get("force_restricted_owner_id", None)
+        if owner_id is None or restricted_owner is None:
             return records
         return records.filtered(
-            lambda q: q.owner_id == (restricted_owner_id or self.env["res.partner"])
+            lambda q: q.owner_id == (restricted_owner or self.env["res.partner"])
         )
 
     @api.model
     def read_group(
         self, domain, fields, groupby, offset=0, limit=None, orderby=False, lazy=True
     ):
-        restricted_owner_id = self.env.context.get("force_restricted_owner_id", None)
-        if restricted_owner_id is not None:
-            domain = expression.AND([domain, [("owner_id", "=", restricted_owner_id)]])
+        owner_in_domain = any(t[0] == "owner_id" for t in domain)
+        if not owner_in_domain:
+            restricted_owner = self.env.context.get("force_restricted_owner_id", None)
+            if restricted_owner is not None:
+                domain = expression.AND(
+                    [domain, [("owner_id", "=", restricted_owner.id)]]
+                )
```

Comment:
I just realized this changes is not valid. I think this changes will lead to error when we search the quantity only for no owner_id (https://github.com/OCA/stock-logistics-workflow/blob/4b0301ca409ade61958fd09d50a10209d064b6c5/stock_owner_restriction/models/product.py#L27-L36). The intention of force_restricted_owner_id is ID or False from somewhere.

Reply:
I can fix but I want to know is there is any reason to change from `restricted_owner_id` to `restricted_owner`? If not, I will revert it and make adjustments.

Reply:
> I can fix but I want to know is there is any reason to change from `restricted_owner_id` to `restricted_owner`? If not, I will revert it and make adjustments.

Due to the convention. `xxx_id` as a variable usually represents an id (integer), which is not the case here.

Reply:
I know the variable name is the case here. But force_restricted_owner_id is an integer or False(for getting no owner quant). So, we should keep using `restricted_owner_id` for avoiding misunderstanding because the value is not record.